### PR TITLE
Add description field to metrics in Pinot

### PR DESCRIPTION
**Pull request by navina**
This PR introduces a new `description` field in Pinot's metric abstraction model. 

* This allows a developer to add metrics related description/documentation directly in the code
* We can further automate metrics documentation generation using a standalone script

### CODE REVIEW

**Comment by navina** [comment]
@xiangfu0 /  @Jackie-Jiang : Review pls!

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java [hunk 1]
```diff
@@ -54,10 +54,16 @@ public enum ServerGauge implements AbstractMetrics.Gauge {
   private final String _unit;
   private final boolean _global;
 
+  private final String _description;
+
   ServerGauge(String unit, boolean global) {
+    this(unit, global, "");
+  }
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java:61]: nit: new line

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerMeter.java [hunk 2]
```diff
@@ -101,11 +101,16 @@ public enum ServerMeter implements AbstractMetrics.Meter {
   private final String _meterName;
   private final String _unit;
   private final boolean _global;
+  private final String _description;
 
   ServerMeter(String unit, boolean global) {
+    this(unit, global, "");
+  }
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerMeter.java:108]: Nit: new line

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerMetrics.java [hunk 3]
```diff
@@ -79,4 +79,10 @@ protected ServerMeter[] getMeters() {
   protected ServerGauge[] getGauges() {
     return ServerGauge.values();
   }
+
+  public static void main(String[] args) {
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerMetrics.java:83]: Remove this

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerTimer.java [hunk 4]
```diff
@@ -67,4 +76,16 @@ public String getTimerName() {
   public boolean isGlobal() {
     return _global;
   }
+
+  @Override
+  public String getDescription() {
+    return _description;
+  }
+
+  public static void main(String[] args) {
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerTimer.java:85]: Remove this

**Review by xiangfu0 — APPROVED** [review]

**Review by gortiz — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java [hunk 5]
```diff
@@ -93,6 +93,10 @@ public PinotMetricsRegistry getMetricsRegistry() {
 
   public interface QueryPhase {
     String getQueryPhaseName();
+
+    default String getDescription() {
+      return "";
+    }
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java:99]: why do we need the default behavior if the constructor is already catching and setting `""` ? ( are we concern backward-compatibility for other extensions?)

**Review by walterddr — APPROVED** [review]
lgtm. minor comment

**Review by npawar — COMMENTED** [review]
Id auto documentation generation the primary motivation? (Not for this PR, general question) Is there a way to extract that field in our prometheus config files, so it can be used in our observability tools?

**Comment by navina** [comment]
> Is auto documentation generation the primary motivation? (Not for this PR, general question) Is there a way to extract that field in our prometheus config files, so it can be used in our observability tools?

documentation and auto-generation of monitoring files (like the ones we use with prometheus export). But that requires a little bit more work to get there. In the mean time, I wanted to kickstart the practice of making documentation as a part of the code.

**Pull request merged**